File: src/test/java/io/binac/leetcode/WordLadderTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class WordLadderTest {
    private final WordLadder.Solution1 solution1 = new WordLadder.Solution1();

    @Test
    void test1() {
        List<String> wordList1 = Arrays.asList("hot", "dot", "dog", "lot", "log", "cog");
        assertEquals(5, solution1.ladderLength("hit", "cog", wordList1));

        List<String> wordList2 = Arrays.asList("hot", "dot", "dog", "lot", "log");
        assertEquals(0, solution1.ladderLength("hit", "cog", wordList2));
    }
}

File: cpp/unittests/WordLadderTest.cpp
#include "WordLadder.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(WordLadder, Solution127_1) {
  const std::string BeginWord1 = "hit", EndWord1 = "cog";
  std::vector<std::string> WordList1{
      "hot", "dot", "dog", "lot", "log", "cog"
  };
  const int Expected1 = 5;
  EXPECT_EQ(Expected1,
            Solution127_1().ladderLength(BeginWord1, EndWord1, WordList1));

  const std::string BeginWord2 = "hit", EndWord2 = "cog";
  std::vector<std::string> WordList2{
      "hot", "dot", "dog", "lot", "log"
  };
  const int Expected2 = 0;
  EXPECT_EQ(Expected2,
            Solution127_1().ladderLength(BeginWord2, EndWord2, WordList2));
}